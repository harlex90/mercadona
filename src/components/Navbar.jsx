import { useIsAuthenticated, useSignOut, useSignIn } from "react-auth-kit";
import { Link, useNavigate } from "react-router-dom";

const Navbar = () => {
    const isAuthenticated = useIsAuthenticated();
    const signout = useSignOut()
    const navigate = useNavigate();

    const onSignin = () => {
        navigate("/loginadmin")
    }

    const onSignout = () => {
       signout();
       navigate('/')
    }
    const navElements = [{title: "Home", id:1, link: "/home"},
                        {title: "Catalogue", id: 2, link: "/catalogue"},
                        {title: "About", id: 3, link: "/about"}]

                        

    const listNavElements = navElements.map(element =>
    <li key={element.id}>
        <Link to={element.link} style={{textDecoration: "none", fontWeight: "bold"}}>
            {element.title}
        </Link>
    </li>
    );
    return(
            
        <ul style={{display: "flex", justifyContent: "center", listStyleType: "none", margin: 0, padding: 0, gap: "100px", alignItems: "center"}}>
            {listNavElements}
            {isAuthenticated() && <li><a style={{textDecoration: "none", fontWeight: "bold"}} href="#" onClick={onSignout}>Signout</a></li>}
            {!isAuthenticated() && <li><a style={{textDecoration: "none", fontWeight: "bold", cursor: "pointer"}} href="#" onClick={onSignin}>Sign In</a></li>}
        </ul>
    
    )                    
}

export default Navbar;