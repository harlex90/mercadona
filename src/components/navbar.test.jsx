import { render } from '@testing-library/react';
import Navbar from './Navbar';

jest.mock('react-auth-kit', () => ({
    useIsAuthenticated: () => () => false, // later, we will change implementation of this mock BY TEST
    useSignOut: () => () => console.log('signOut()')
}))

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    Link: ({ to, children }) => <div data-testid="Link" data-to={to}>{children}</div>,
    useNavigate: () => mockNavigate
}))

describe('Navbar', () => {
    test('match snapshot', () => {
        const { asFragment } = render(<Navbar />);
        expect(asFragment()).toMatchSnapshot()
    });

    test('should navigate to /loginadmin if click button signin', () => {
        const { getByText } = render(<Navbar />);
        const buttonSignin = getByText('Sign In');
        buttonSignin.click();
        expect(mockNavigate).toHaveBeenCalledWith("/loginadmin")
    });
    // test('display menu and sign in/sign out only if admin is connected', () => {
    //     const { getByText } = render(<Navbar />);
    //     fireEvent.click(screen.getByText('Home'))
    //     fireEvent.click(screen.getByText('Catalogue'))
    //     fireEvent.click(screen.getByText('About'))
    //     fireEvent.click(screen.getByText('Sign in'))
    //     fireEvent.click(screen.getByText('Sign out'))
    //     const myButton = getByText('Click');
       
        
    // });

    
});

