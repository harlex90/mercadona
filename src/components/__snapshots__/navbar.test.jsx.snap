// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar match snapshot 1`] = `
<DocumentFragment>
  <ul
    style="display: flex; justify-content: center; list-style-type: none; margin: 0px; padding: 0px; gap: 100px; align-items: center;"
  >
    <li>
      <div
        data-testid="Link"
        data-to="/home"
      >
        Home
      </div>
    </li>
    <li>
      <div
        data-testid="Link"
        data-to="/catalogue"
      >
        Catalogue
      </div>
    </li>
    <li>
      <div
        data-testid="Link"
        data-to="/about"
      >
        About
      </div>
    </li>
    <li>
      <a
        href="#"
        style="text-decoration: none; font-weight: bold; cursor: pointer;"
      >
        Sign In
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
